main.go

```diff
@@ -80,6 +90,7 @@ func runMain() error {
 		// Start or stop recording if required.
 		if recordingCount > 0 && writer == nil {
 			filename := filepath.Join(conf.OutputDir, newRecordingTempName())
+			defer os.Remove(filename)
```

Comment:
Similarly, this is going to build up a bunch of defers (which consume memory) until `runRecordings` returns. 

A better solution to removing an in-progress video file when `runRecordings` exits would be to put something like this just before the for loop:

```go
defer func() {
    if writer != nil {
        writer.Close()
        os.Remove(writer.Name())
    }
}()
```